### ISSUE TITLE
Editing keys

### ISSUE THREAD

**Issue by pimterry** [mac-s-g/react-json-view#182]
Is there any way to edit a key? It'd be really useful to be able to rename properties.

I see this was mentioned in #91, but it looked like it got closed without this being added, as far as I can tell.

**Comment by pimterry** [comment]
@mac-s-g makes sense! I'm imagining something pretty close to the current implementation I think:

* An extra button when mousing over keys, i.e. there should be an edit button on the `object` line here:  
  ![screenshot from 2018-04-16 13-28-20](https://user-images.githubusercontent.com/1526883/38806637-284ae980-417a-11e8-9ebd-fd4741955e02.png)
  When clicked, the key should become a text box, with save/cancel buttons following it (you don't need the type confirmation button that values have, since it's always a string)

* When editing a value, the key should also become editable, i.e. here there should be two text boxes, not one:
  ![screenshot from 2018-04-16 13-28-32](https://user-images.githubusercontent.com/1526883/38806656-38a7f106-417a-11e8-9ddd-e86c198dceac.png)
  (But I think you still only need the one set of save controls - you could save/cancel them together, that's fine)

**Comment by brentluna** [comment]
Any update on this?

**Comment by sushantarsenal** [comment]
In the current version, is it possible to replace the textbox with a dropdown(with a list of my own options to select from) while trying to edit values?
<img width="376" alt="screen shot 2018-09-18 at 12 42 14 pm" src="https://user-images.githubusercontent.com/8470688/45669623-539e8980-bb40-11e8-9667-d8b013e5dc35.png">
For example, in this case, I would like to open a dropdown(with Male and Female options) instead of a textbox.

**Comment by mateosilguero** [comment]
I made a solution for this here: https://github.com/mac-s-g/react-json-view/pull/233

**Comment by rathish** [comment]
> In the current version, is it possible to replace the textbox with a dropdown(with a list of my own options to select from) while trying to edit values?
> <img alt="screen shot 2018-09-18 at 12 42 14 pm" width="376" src="https://user-images.githubusercontent.com/8470688/45669623-539e8980-bb40-11e8-9667-d8b013e5dc35.png">
> For example, in this case, I would like to open a dropdown(with Male and Female options) instead of a textbox.

I created quick solution, to render textarea or select based initiValues property.  [check](https://github.com/rathish/react-json-view/commit/960f91755620db443c9d080eee9e64244ea85fe2)

**Comment by hari4125862** [comment]
Hey any update on the about feature?

**Comment by sriramarunachalam** [comment]
Is there any update on this issue?

**Comment by mnavon** [comment]
Is there any update on this feature?
